app/controllers/placements/schools/placements_controller.rb

```diff
@@ -46,6 +52,28 @@ def set_decorated_placement
     @placement = @placement.decorate
   end
 
+  def valid_attributes?
+    if params[:edit_path] == "edit_provider"
+      placement_params[:provider_id].present?
+    else # params[:edit_path] == "edit_mentors"
+      valid_mentor_ids?(placement_params[:mentor_ids].compact_blank)
+    end
+  end
+
+  def valid_mentor_ids?(mentor_ids)
+    if mentor_ids.include?("not_known")
+      params[:placement][:mentor_ids] = []
+      return true
+    end
+
+    if mentor_ids.present? && mentor_ids.all? { |id| Placements::Mentor.exists?(id:) && @school.mentors.exists?(id:) }
```

Comment:
Do we need both `exists?` calls here? I think the latter should probably cover it.

```suggestion
    if mentor_ids.present? && mentor_ids.all? { |id| @school.mentors.exists?(id:) }
```